src/apps/relay/mainrelay.c

```diff
@@ -2779,1082 +2968,1086 @@ static void print_features(unsigned long mfn) {
+  }
+#endif
 
-    free(full_path_to_file);
-    return;
+  if (!(turn_params.no_tls && turn_params.no_dtls) && !turn_params.cert_file[0]) {
+    TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING, "\nWARNING: certificate file is not specified, I cannot start TLS/DTLS "
+                                          "services.\nOnly 'plain' UDP/TCP listeners can be started.\n");
+    turn_params.no_tls = 1;
+    turn_params.no_dtls = 1;
   }
 
-keyerr : {
-  if (critical) {
+  if (!(turn_params.no_tls && turn_params.no_dtls) && !turn_params.pkey_file[0]) {
+    TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING, "\nWARNING: private key file is not specified, I cannot start TLS/DTLS "
+                                          "services.\nOnly 'plain' UDP/TCP listeners can be started.\n");
     turn_params.no_tls = 1;
     turn_params.no_dtls = 1;
-    TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING, "cannot start TLS and DTLS listeners because %s file is not set properly\n",
-                  file_title);
   }
-  if (full_path_to_file)
-    free(full_path_to_file);
-  return;
-}
-}
 
-static void adjust_key_file_names(void) {
-  if (turn_params.ca_cert_file[0])
-    adjust_key_file_name(turn_params.ca_cert_file, "CA", 1);
-  adjust_key_file_name(turn_params.cert_file, "certificate", 1);
-  adjust_key_file_name(turn_params.pkey_file, "private key", 1);
-  if (turn_params.dh_file[0])
-    adjust_key_file_name(turn_params.dh_file, "DH key", 0);
-}
-static DH *get_dh566(void) {
+  if (!(turn_params.no_tls && turn_params.no_dtls)) {
+    adjust_key_file_names();
+  }
 
-  unsigned char dh566_p[] = {0x36, 0x53, 0xA8, 0x9C, 0x3C, 0xF1, 0xD1, 0x1B, 0x2D, 0xA2, 0x64, 0xDE, 0x59, 0x3B, 0xE3,
-                             0x8C, 0x27, 0x74, 0xC2, 0xBE, 0x9B, 0x6D, 0x56, 0xE7, 0xDF, 0xFF, 0x67, 0x6A, 0xD2, 0x0C,
-                             0xE8, 0x9E, 0x52, 0x00, 0x05, 0xB3, 0x53, 0xF7, 0x1C, 0x41, 0xB2, 0xAC, 0x38, 0x16, 0x32,
-                             0x3A, 0x8E, 0x90, 0x6C, 0x7E, 0xD1, 0x44, 0xCB, 0xF9, 0x2D, 0x1E, 0x4A, 0x9A, 0x32, 0x81,
-                             0x58, 0xE1, 0xE1, 0x17, 0xC1, 0x9C, 0xF1, 0x1E, 0x96, 0x2D, 0x5F};
+  openssl_load_certificates();
 
-  //	-----BEGIN DH PARAMETERS-----
-  // MEwCRzZTqJw88dEbLaJk3lk744wndMK+m21W59//Z2rSDOieUgAFs1P3HEGyrDgW
-  // MjqOkGx+0UTL+S0eSpoygVjh4RfBnPEeli1fAgEF
-  //	-----END DH PARAMETERS-----
+  if (turn_params.verbose)
+    TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "no_tls:%d\tno_tlsv1:%d\tno_tlsv1_1:%d\tno_tlsv1_2:%d\tno_dtls:%d\n",
+                  turn_params.no_tls, turn_params.no_tlsv1, turn_params.no_tlsv1_1, turn_params.no_tlsv1_2,
+                  turn_params.no_dtls);
+}
 
-  unsigned char dh566_g[] = {0x05};
-  DH *dh;
+static void openssl_load_certificates(void) {
 
-  if ((dh = DH_new()) == NULL)
-    return (NULL);
+  print_abs_file_name("", "Certificate", turn_params.cert_file);
+  print_abs_file_name("", "Private key", turn_params.pkey_file);
+
+  TURN_MUTEX_LOCK(&turn_params.tls_mutex);
+  if (!turn_params.no_tls) {
```

Comment:
boy it's really awkward to see ` ! turn_params.no_tls` over and over.

We should seriously consider making this say `turn_params.tls_enabled` instead, so that the code doesn't need to negate every place that it checks it.